client/src/api.ts

```diff
@@ -78,6 +79,28 @@ graphqlRequest.interceptors.response.use(
   }
 );
 
+interface GQLResponse<T> {
+  errors?: { message: string }[];
+  data?: T | { me: T };
+}
+function getGqlReponseData<T>(res: AxiosResponse<GQLResponse<T>>): T {
+  if (res.status !== 200) {
+    throw new Error(
+      `GQL HTTP request failed with status ${res.status}: ${res.statusText}`
+    );
+  }
+  if (res.data.errors) {
+    throw new Error(
+      `errors in GQL response: ${JSON.stringify(res.data.errors)}`
+    );
+  }
+  if (!res.data.data) {
+    throw new Error(`no data in response: ${JSON.stringify(res.data)}`);
+  }
+  if ("me" in res.data.data) return res.data.data.me;
```

Comment:
the whole `me` thing makes some misery, but we may as well deal w it correctly, e.g.

```typescript
interface Me<T> {
  me: T
}
```